lab/runnerctl.py

```diff
@@ -32,13 +31,23 @@ def pytest_namespace():
     return {'halt': Halt()}
 
 
+def pytest_configure(config):
+    signal.signal(signal.SIGTERM, exit_gracefully)
+
+
 @pytest.hookimpl(hookwrapper=True)
 def pytest_runtest_protocol(item, nextitem):
     yield
     if pytest.halt.msg:
         item.session.shouldstop = pytest.halt.msg
 
 
+def pytest_runtest_makereport(item, call):
+    if 'setup_test' in item.keywords and call.excinfo:
```

Comment:
Yeah, actually, this needs some work. As much as it works, it swallows the error that triggers it...

I want to contribute this upstream and see if they'll take it.